ui/app/src/utils/content.ts

```diff
@@ -1147,6 +1163,83 @@ export function generatePlaceholderImageDataUrl(attributes?: Partial<GeneratePla
 	return canvas.toDataURL();
 }
 
+/**
+ * Retrieves the value of a property from a content instance model, considering system properties.
+ *
+ * @param model - The content instance model to retrieve the value from.
+ * @param prop - The property to retrieve the value from.
+ * @returns The value of the property from the model.
+ * */
+export function getContentInstanceValueFromProp(model: ContentInstance, prop: string) {
+	const systemProp = systemPropMap[prop];
+	if (systemProp) {
+		if (systemProp === 'fileName') {
+			return getContentInstanceFileName(model);
+		} else {
+			return model.craftercms[systemProp];
+		}
+	} else {
+		return model[prop];
+	}
+}
+
+export const systemPropToXmlMap = {
+	fileName: 'file-name',
+	internalName: 'internal-name'
+};
+
+export function getContentInstanceXmlValueFromProp(xml: string, prop: string): string {
+	const selectionProp = systemPropToXmlMap[prop] ?? prop;
+	const doc = fromString(xml).querySelector(selectionProp);
+	return doc ? serialize(doc) : '';
+}
+
+export function getContentFileNameFromPath(path: string): string {
+	let fileName = path.replace('/site/website', '');
+
+	if (path.endsWith('/index.xml')) {
+		fileName = fileName.replace('/index.xml', '');
+		return fileName.substring(fileName.lastIndexOf('/')) || '/';
+	}
+
+	if (path.includes('.xml')) {
+		return fileName.substring(fileName.lastIndexOf('/') + 1).replace('.xml', '');
+	}
+	return fileName;
+}
+
+/**
+ * Retrieves the file name from the given content instance model.
+ *
+ * @param model - The content instance model containing the path.
+ * @returns The file name extracted from the model's path, or null if the path is not defined.
+ */
+export function getContentInstanceFileName(model: ContentInstance) {
+	const path = model.craftercms.path;
+	if (!path) return null;
+
+	return getContentFileNameFromPath(path);
+}
+
+export const mockContentInstance = {
+	craftercms: {
+		id: null,
+		path: null,
+		label: null,
+		dateCreated: null,
+		dateModified: null,
+		contentTypeId: null,
+		disabled: false
+	}
+};
```

Comment:
Might be best to make this into a function to keep the object off memory and avoid it getting modified. On contentController, guest-side, I think something like this is produced when adding new components. Pls check and see if we should be refactoring something.